tests/test_getRaw.py

```diff
@@ -158,7 +158,7 @@ def testFringe(self):
     def testDefect(self):
         """Test retrieval of defect list"""
         defectList = self.butler.get("defects", self.dataId)
-        self.assertEqual(len(defectList), 9)
+        self.assertEqual(len(defectList), 408)
```

Comment:
I see 300-someodd individual defects in the new file for ccd S29 (ccdnum 1), which varies a bit depending on which date/version I choose.

Two questions: Why was the value previously 9 and where is the new 408 coming from? What mechanism (if any) exists to specify the desired defect version (date) during defect ingestion, and can that be tested here?

Reply:
The data is in the dataId.  As far as I know, this should have been selecting the same defects, but the version used in the test was one that was special for the test.  It could have been truncated for all I know.  The 408 number is just the length of the defect list for that dataId: ccd and date.

Reply:
When you say "the dat[e] is in the dataId", I guess what I am really wondering is what a user should type during defect ingestion to say "I want to ingest and use defects from 2013" vs "I want to ingest and use defects from 2015." Ingestion is kind of a pre-dataId operation in my mind. Is the idea to ingest **all** available defects with non-overlapping validities and then assign the appropriate one based on the science image's dataId? If yes, I'm happy. If something else happens, please explain.

Reply:
The metadata contains the valid start date and the assumption is that the bad pixel mask is valid until the next bad pixel mask comes along.  That is what defines the validity range.  So the user need do nothing except point `ingestDefects.py` at a properly formatted tree of defects files as provided in `obs_decam_data`.